### ISSUE TITLE
Using Zipkin exporter in open-telemetry plugin for Playwright Engine completely prevents vUsers from failing

### ISSUE THREAD

**Issue by up201306506** [artilleryio/artillery#2365]
Version info: 2.0.3

Create test that's **guaranteed** to fail

```js
const { expect } = require('@playwright/test');
async function RunTest(page, userContext, events, test) {
    await test.step("Must Fail", async () => {
        await expect(false).toBeTruthy();
    });
}
module.exports = { RunTest };
```

Import it onto Artillery file using the Playwright engine. Also setup Zipkin's exporter. Run a load test.

I expected to see this happen:

```
vusers.created: ................................................................ 5
vusers.created_by_name.LoadTestFunction: ....................................... 5
vusers.failed: ................................................................. 5

Checks:
fail: (vusers.failed / vusers.created) < 0.5
```

Instead, this happened:

```
vusers.created: ................................................................ 5
vusers.created_by_name.LoadTestFunction: ....................................... 5
vusers.failed: ................................................................. 0
vusers.session_length:
  min: ......................................................................... 782.1
  max: ......................................................................... 1037.5
  mean: ........................................................................ 875.1
  median: ...................................................................... 837.3
  p95: ......................................................................... 889.1
  p99: ......................................................................... 889.1

Checks:
ok: (vusers.failed / vusers.created) < 0.5
```

Removing code from the `test.step()` function fixes the issue. 
Commenting out the open-telemetry segment in artillery config also fixes the issue.  
Bug is exclusive to open-telemetry, as I have a prometheus pushgate side to the plug-in operating concurrently with no change to either isolated or joint behavior.

Adding problem code outside of a step function reintroduces fail states, but only on the code that's outside function
```js
await test.step("Must Fail", async () => {
    await expect(false).toBeTruthy();
});
await expect(false).toBeTruthy();
```
```
Ge [Error]: expect(received).toBeTruthy()

Received: false
    at Proxy.<anonymous> (C:\...\node_modules\playwright\lib\matchers\expect.js:172:37)
    at RunTest (C:\...\test.js:10:25)  # ----> This is the second assertion

vusers.created: ................................................................ 5
vusers.created_by_name.LoadTestFunction: ....................................... 5
vusers.failed: ................................................................. 5

Checks:
fail: (vusers.failed / vusers.created) < 0.5
```
In other words, this bug is directly caused by the handling of `test.step()` by the plug-in. 

Occurring both locally on Windows and on external Alpine containers
Unsure if this is isolated to zipkin or to open-telemetry on the whole as I've not had success exporting to a Otel Collector Receiver with other protocols, but that's a separate issue.

**Comment by InesNi** [comment]
Hi @up201306506,

Thank you for reporting this, and for all the details! 🙇🏻‍♀️ 
I will take a look and get back to you when I know more.

**Comment by InesNi** [comment]
Hi @up201306506 👋🏻,

A fix for this issue is available in the following canary: 2.0.3-67408d1
It will also be available in the next version release.

Thank you once again for reporting this and providing all the details! 🙌🏻
If you need help around sending data to the Collector, or have any other questions feel free to open a discussion or another issue!

Happy testing 🚀
